meson.build

```diff
@@ -0,0 +1,484 @@
+    env: test_env,
+)
+test(
+    'keymap',
+    executable('test-keymap', 'test/keymap.c', dependencies: test_dep),
+    env: test_env,
+)
+test(
+    'filecomp',
+    executable('test-filecomp', 'test/filecomp.c', dependencies: test_dep),
+    env: test_env,
+)
+test(
+    'context',
+    executable('test-context', 'test/context.c', dependencies: test_dep),
+    env: test_env,
+)
+test(
+    'rules-file',
+    executable('test-rules-file', 'test/rules-file.c', dependencies: test_dep),
+    env: test_env,
+)
+test(
+    'stringcomp',
+    executable('test-stringcomp', 'test/stringcomp.c', dependencies: test_dep),
+    env: test_env,
+)
+test(
+    'buffercomp',
+    executable('test-buffercomp', 'test/buffercomp.c', dependencies: test_dep),
+    env: test_env,
+)
+test(
+    'log',
+    executable('test-log', 'test/log.c', dependencies: test_dep),
+    env: test_env,
+)
+test(
+    'atom',
+    executable('test-atom', 'test/atom.c', dependencies: test_dep),
+    env: test_env,
+)
+test(
+    'utf8',
+    executable('test-utf8', 'test/utf8.c', dependencies: test_dep),
+    env: test_env,
+)
+test(
+    'state',
+    executable('test-state', 'test/state.c', dependencies: test_dep),
+    env: test_env,
+)
+test(
+    'keyseq',
+    executable('test-keyseq', 'test/keyseq.c', dependencies: test_dep),
+    env: test_env,
+)
+test(
+    'rulescomp',
+    executable('test-rulescomp', 'test/rulescomp.c', dependencies: test_dep),
+    env: test_env,
+)
+test(
+    'compose',
+    executable('test-compose', 'test/compose.c', dependencies: test_dep),
+    env: test_env,
+)
+test(
+    'symbols-leak-test',
+    find_program('test/symbols-leak-test.bash'),
+    env: test_env,
+)
+if get_option('enable-x11')
+    test(
+        'x11',
+        executable('test-x11', 'test/x11.c', dependencies: test_dep, link_with: libxkbcommon_x11_internal),
+        env: test_env,
+    )
+    # test/x11comp is meant to be run, but it is (temporarily?) disabled.
+    # See: https://github.com/xkbcommon/libxkbcommon/issues/30
+    executable('test-x11comp', 'test/x11comp.c', dependencies: test_dep, link_with: libxkbcommon_x11_internal)
+endif
+
+
+# Demo programs.
+executable('rmlvo-to-kccgst', 'test/rmlvo-to-kccgst.c', dependencies: test_dep)
+executable('print-compiled-keymap', 'test/print-compiled-keymap.c', dependencies: test_dep)
+if cc.has_header('linux/input.h')
+    executable('interactive-evdev', 'test/interactive-evdev.c', dependencies: test_dep)
+endif
+if get_option('enable-x11')
+    executable('interactive-x11', 'test/interactive-x11.c', dependencies: test_dep, link_with: libxkbcommon_x11_internal)
+endif
+if get_option('enable-wayland')
+    wayland_client_dep = dependency('wayland-client', version: '>=1.2.0')
+    wayland_protocols_dep = dependency('wayland-protocols', version: '>=1.7')
+    wayland_scanner_dep = dependency('wayland-scanner')
+    if not wayland_client_dep.found() or not wayland_protocols_dep.found() or not wayland_scanner_dep.found()
```

Comment:
Same comment as on the X11 deps, except these ones don't actually have required: false, so you'll never hit this branch.